src/test/java/org/sonatype/goodies/packageurl/PackageUrlBuilderTest.groovy

```diff
@@ -57,71 +77,112 @@ class PackageUrlBuilderTest
     List<String> ns = ['a', 'b', 'c']
-    PackageUrl purl = new PackageUrlBuilder().type('foo').name('bar').namespace(ns).build()
-    assert purl.namespace == ns
+    def builder = new PackageUrlBuilder().type('foo').name('bar').namespace(ns)
+    assert builder.build().namespace == ns
+    assert builder.asIsMode(true).build().namespace == ns
   }
 
   @Test
   void 'namespace parsed'() {
     List<String> ns = ['a', 'b', 'c']
-    PackageUrl purl = new PackageUrlBuilder().type('foo').name('bar').namespace(ns.join('/')).build()
-    assert purl.namespace == ns
+    def builder = new PackageUrlBuilder().type('foo').name('bar').namespace(ns.join('/'))
+    assert builder.build().namespace == ns
+    assert builder.asIsMode(true).build().namespace == ns
   }
 
   @Test
   void 'subpath asis'() {
     List<String> subpath = ['a', 'b', 'c']
-    PackageUrl purl = new PackageUrlBuilder().type('foo').name('bar').subpath(subpath).build()
-    assert purl.subpath == subpath
+    def builder = new PackageUrlBuilder().type('foo').name('bar').subpath(subpath)
+    assert builder.build().subpath == subpath
+    assert builder.asIsMode(true).build().subpath == subpath
   }
 
   @Test
   void 'subpath parsed'() {
     List<String> subpath = ['a', 'b', 'c']
-    PackageUrl purl = new PackageUrlBuilder().type('foo').name('bar').subpath(subpath.join('/')).build()
-    assert purl.subpath == subpath
+    def builder = new PackageUrlBuilder().type('foo').name('bar').subpath(subpath.join('/'))
+    assert builder.build().subpath == subpath
+    assert builder.asIsMode(true).build().subpath == subpath
   }
 
   @Test
   void 'qualifiers append'() {
     PackageUrlBuilder builder = new PackageUrlBuilder().type('foo').name('bar')
-    builder.qualifiers([a: '1', b: '2'])
+    builder.qualifiers([a: '1', B: '2'])
     builder.qualifier('c', '3')
     builder.build().with {
       assert qualifiers == [a: '1', b: '2', c: '3']
     }
+    builder.asIsMode(true).build().with {
+      assert qualifiers == [a: '1', b: '2', c: '3']
+    }
   }
 
   @Test
   void 'qualifiers null resets'() {
     PackageUrlBuilder builder = new PackageUrlBuilder().type('foo').name('bar')
     builder.qualifier('a', '1')
     builder.qualifiers((Map)null)
-    builder.qualifier('b', '2')
+    builder.qualifier('B', '2')
     builder.build().with {
       assert qualifiers == [b: '2']
     }
+    builder.asIsMode(true).build().with {
+      assert qualifiers == [b: '2']
+    }
   }
 
   @Test
   void 'qualifiers that are empty'() {
     PackageUrlBuilder builder = new PackageUrlBuilder().type('foo').name('bar')
-    builder.qualifiers([a: '1', b: ''])
+    builder.qualifiers([A: '1', b: ''])
     builder.qualifier('c', '3')
     builder.qualifier('d', '')
     builder.build().with {
       assert qualifiers == [a: '1', c: '3']
     }
+    builder.asIsMode(true).build().with {
+      assert qualifiers == [a: '1', c: '3']
+    }
+  }
+
+  @Test
+  void 'bitbucket namespace and name'() {
```

Comment:
should we have a similar test for github?

Reply:
Will do